Solutions/Corvus.Json.ExtendedTypes/Corvus.Json/JsonAny.cs

```diff
@@ -504,12 +511,40 @@ public JsonBoolean AsBoolean
         public JsonNull AsNull
 #pragma warning restore CA1822 // Mark members as static
         {
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
             get
             {
                 return default;
             }
         }
 
+        /// <summary>
+        /// Gets the instance as a list of <see cref="JsonAny"/>.
+        /// </summary>
+        public ImmutableList<JsonAny> AsItemsList
+        {
+            get
+            {
+                if (this.arrayBacking is ImmutableList<JsonAny> items)
+                {
+                    return items;
+                }
+
+                if (this.jsonElementBacking.ValueKind == JsonValueKind.Array)
+                {
+                    ImmutableList<JsonAny>.Builder builder = ImmutableList.CreateBuilder<JsonAny>();
+                    foreach (JsonElement item in this.jsonElementBacking.EnumerateArray())
+                    {
+                        builder.Add(new JsonAny(item));
+                    }
+
+                    return builder.ToImmutable();
+                }
+
+                return ImmutableList<JsonAny>.Empty;
```

Comment:
I don't think it is necessarily a good idea - but it is a significant breaking change so I opened an issue a while back.

I have tried to avoid throwing exceptions in most codepaths. However, in this kind of case, you can check the `ValueKind` and guarantee it won't throw - so it's on you if you cause an exception. So I *think* the right thing to do is throw, as you imply.

This was captured in 
https://github.com/corvus-dotnet/Corvus.JsonSchema/issues/31